packages/cli/internal/pkg/cli/context_deploy.go

```diff
@@ -89,45 +84,94 @@ func (o *deployContextOpts) validateSuppliedContexts(contextList []string) error
 // Execute causes the specified context(s) to be deployed.
 func (o *deployContextOpts) Execute() ([]context.Detail, error) {
 	o.contexts = slices.DeDuplicateStrings(o.contexts)
-	results := o.deployContexts(o.contexts)
-	contextDetails := make([]context.Detail, len(results))
-	hasErrors := false
+
+	err := o.deployContexts()
+	if err != nil {
+		return nil, err
+	}
+
+	contextDetails, err := o.validateDeploymentResults()
+	if err != nil {
+		return nil, err
+	}
+
+	sortContextDetails(contextDetails)
+	return contextDetails, nil
+}
+
+func (o *deployContextOpts) deployContexts() error {
+	progressResults := o.ctxManagerFactory().Deploy(o.contexts)
 	aggregateSuggestions := make([]string, 0)
-	for i, result := range results {
-		if result.err != nil {
+
+	failedDeployments := make([]context.ProgressResult, 0)
+	for _, progressResult := range progressResults {
+		if progressResult.Err != nil {
+			failedDeployments = append(failedDeployments, progressResult)
+		}
+	}
+
+	failedDeploymentsLength := len(failedDeployments)
+	if failedDeploymentsLength > 0 {
+		for i, failedDeployment := range failedDeployments {
+			log.Error().Msgf("Failed to deploy context '%s'. Below is the log for that deployment", failedDeployment.Context)
+
+			outputsLength := len(failedDeployment.Outputs)
+			for i := 0; i < outputsLength-1; i++ {
+				log.Error().Msg(failedDeployment.Outputs[i])
```

Comment:
If 3/3 deployments fail are we going to dump all the CDK logs into the terminal? That seems unmanageable to users. Half of it would probably get cut off due to terminal window limits. Perhaps if `failedDeploymentsLength > 1` we direct users to CloudFormation?